nats-base-client/kv.ts

```diff
@@ -367,13 +368,20 @@ export class Bucket implements KV, KvRemove {
     return pa.seq;
   }
 
-  async get(k: string): Promise<KvEntry | null> {
+  async get(
+    k: string,
+    opts: { revision: number } = { revision: 0 },
+  ): Promise<KvEntry | null> {
     const ek = this.encodeKey(k);
     this.validateKey(ek);
+
+    let arg: MsgRequest = { last_by_subj: this.fullKeyName(ek) };
+    if (opts.revision !== 0) {
+      arg = { seq: opts.revision };
+    }
+
     try {
-      const sm = await this.jsm.streams.getMessage(this.bucketName(), {
-        last_by_subj: this.fullKeyName(ek),
-      });
+      const sm = await this.jsm.streams.getMessage(this.bucketName(), arg);
```

Comment:
There is a nuance in the Go client: when asking for a revision (sequence number), it is possible that a message is found at that sequence, but that does not mean that it will match the subject (the key). So if you had "A"/seq 1, "B"/seq 2 and you ask Get("A", 2), a message will be returned by the low level "getMessage(2)", but that does not match the key/subject "A", so you should return "key not found" in that case.